control-plane/src/modules/runs/agent/tool.ts

```diff
@@ -55,3 +57,59 @@ export class AgentTool {
     }
   }
 }
+
+export class AgentToolV2 {
```

Comment:
How does this differ from `AgentToo`?

Reply:
`AgentTool` assumes that the internal tool returns the struct { result, resultType, status }, and pushes off the serialization responsibility to the the `func`.

`AgentToolV2` makes `func` work more like a normal service function in a Node.js service. Technically, the `tool.func` should not be entrusted to report the resolution / rejection or the status - I think? The `status` especially should be `success` for any internal tool calls.

Also in the `V1`, `return this.func(input)` can throw and fail execute, whereas we should make execute return a `success` no matter what.

Thoughts?